implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -133,6 +149,21 @@ public SmallRyeHealth getHealthGroups() {
         return getHealth(groupHealthChecks);
     }
 
+    @Experimental("Asynchronous Health Check procedures")
+    public Uni<SmallRyeHealth> getHealthAsync() {
```

Comment:
Suggestion: Instead of distinguishing the async and sync case, you can make sync look async and have a single method:
```
Uni<HealthCheckResponse> syncMadeAsync = Uni.createFrom(() -> sync.call());
```

Note the usage of a supplier which indicates that `sync.call()` is called on subscription (and on every subscription).

Reply:
Can't reuse the sync health calls here as async works with different interface. But this can be useful in tests. Thanks.

Reply:
That's more something for https://github.com/smallrye/smallrye-health/blob/master/implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java#L137. 

Just have 1 getHealth method instead of 2 and consider the synchronous checks, async.